src/main/frontend/components/header.cljs

```diff
@@ -32,25 +37,72 @@
                    (route-handler/go-to-journals!))}
      (ui/icon "home" {:style {:fontSize ui/icon-size}})]))
 
-(rum/defc login
-  [logged?]
-  (when (and (not logged?)
-             (not config/publishing?))
+(rum/defc login < rum/reactive
+  []
+  (let [_ (state/sub :auth/id-token)]
```

Comment:
`(state/sub :auth/id-token)` here is just to make login component reactive on `:auth/id-token`... but the value is not used.
`state/sub` kind of equals to `rum/react`.
not sure if there is a better way to implement this.

Reply:
Ah. It's being reactive for `logged-in?`. It would be helpful if `logged-in?` took id-token as an argument to make their coupling explicit. Should [this component](https://github.com/logseq/logseq/blob/eaba31ed7ffd1d4e7eda2b19fde7ed1fa4b213f6/src/main/frontend/components/sidebar.cljs#L324-L395) also have that `state/sub`? I wonder if that's causing the [inconsistent bug](https://github.com/logseq/logseq/pull/3839#discussion_r829450522) I was seeing